accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,305 @@
+- Feature Name: Confidential Compute Attestation Report
+- Start Date: 2024-01-16
+
+# Summary
+[summary]: #summary
+
+SUSE Manager / Uyuni should help with Confidential Compute Attestation and being able to generate a report page for systems which should run in this mode.
+
+# Motivation
+[motivation]: #motivation
+
+More and more systems run as virtual machines or (public) cloud instances.
+Customers want to be sure that data of these VMs cannot be read by administrators or the cloud provider.
+Confidential Compute is the way to solve this problem.
+But that a VM run in confidential mode needs to be attested and checked on regular base.
+
+SUSE Manager should provide an attestation report for a system.
+Also a history of the past checks should be kept and shown.
+
+The attestation is a process of multiple checks which can also depend on the hardware and used environment.
+SUSE Manager should provide a framework for executing such checks, store the results in the database and show the latest and historic results.
+The number of checks should be easy extendable in future versions.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Confidential Compute Attestation cannot be performed on all systems.
+It depends on the used hardware and environment where the instance is running on.
+In most cases it is not possible to detect if an attestation can be performed or not.
+
+This requires a per system configuration which should contain at least two configuration options:
+
+- Enable/Disable Confidential Compute Attestation (Checkbox)
```

Comment:
If attestation can't be performed on a system, will we check that part at this time when user try to enable this checkbox or will that happen later? And if attestation is possible , Ideally I would only then mark this checkbox as enabled. But I am not sure how feasible is it to implement that way.

Reply:
Currently it seems that we have no chance to find out if an attestation can be done or not.
The administrator need to know that.

Reply:
This is a general "do it or not" decision for the the admin.